jpa/subsystem/src/main/java/org/jboss/as/jpa/classloader/JPADelegatingClassFileTransformer.java

```diff
@@ -44,13 +46,47 @@ public JPADelegatingClassFileTransformer(PersistenceUnitMetadata pu) {
     public byte[] transform(ClassLoader classLoader, String className, Class<?> aClass, ProtectionDomain protectionDomain, byte[] originalBuffer) throws
         IllegalClassFormatException {
         byte[] transformedBuffer = originalBuffer;
```

Comment:
Since you've introduced the new method, it's probably better to only have the logic in one place.  This method can delegate to that one simply by calling `transform(classLoader, className, protectionDomain, ByteBuffer.wrap(originalBuffer))`.  You can factor out a private utility method to 'convert' the `ByteBuffer` to an array which can be used in both methods.

Reply:
Changed as suggested, will push in a minute.  

Is it okay that transform(ClassLoader, String, Class, ProtectionDomain, byte[]), will now throw IlegalStateException instead of IllegalClassFormatException?

Will org.jboss.modules.ClassTransformer need to deal with redefining already loaded classes?  Currently, j.l.i.ClassFileTransformer#transform is passed "Class<?> classBeingRedefined" for redefinition, which I just pass onto the JPA persistence providers (which they probably ignore).

Reply:
No, we do not support redefinition.  Changing the exception is probably fine in this case (we're the ones who catch it after all).